### ISSUE TITLE
Question: What happens if you get too many unauthorized responses?

### ISSUE THREAD

**Issue by Akumati1435** [Bungie-net/api#884]
I figured this is the kind of thing I should ask about before doing what I'm thinking about doing.

I want to cache my tokens in memory, so they are going to go out of date.

There are all sorts of jobs that use the tokens and my website which can launch a ton of threads to grab high level data on all my clans whenever someone loads it.  There is a background job that refreshes the tokens before they will expire.  When the background job refreshes them, all the tokens in memory in all running jobs and the website will be outdated and I'm guessing they'd throw unauthorized http responses.  My code catches the exception, refreshes its tokens from the database, and retries (in the next version).

Basically.. There are a lot of threads going at a time, so when they expire I might have 20+ failures before the tokens are refreshed and the calls are retried with the new tokens.  Is this going to block my api key?  Will the API even care?

**Comment by vthornheart-bng** [comment]
Sorry for the late reply!  I must be losing some notification emails, this is the second one today that I only spotted because I happened to check the overall list!

My advice would be, as long as you're not intentionally sending failures - that is to say, you're not hitting us with so many failures that it could be construed as some attempt at a DDOS or security attack, and you're keeping under the throttle limit - we're okay with failures coming in.  Your API key won't be blocked unless you hit one of those scenarios, which if you're behaving in general (keep under the throttle, don't spam us with false requests in an attempt to bypass security) you should be good to go.

**Comment by Akumati1435** [comment]
Ok.  I'm obviously going to try and minimize the chances of it happening, but it's bound to have an issue eventually.  The cache has a 60 second expiration date on it so if it happens, it will be for 60 seconds max.  It is also saving the last cache update time to a file so I'm reading that in every single time I go to grab a clan's tokens so it will know if something changed (assuming i dont run out of disk space this should prevent 99% of the unauthorized errors).

I havent gotten many throttled responses recently so I think it will be fine.
